addons/payment/static/src/js/express_checkout_form.js

```diff
@@ -64,11 +64,6 @@ publicWidget.registry.PaymentExpressCheckoutForm = publicWidget.Widget.extend({
     _prepareTransactionRouteParams(providerId) {
         return {
             'payment_option_id': parseInt(providerId),
-            'reference_prefix': this.txContext.referencePrefix &&
-                                this.txContent.referencePrefix.toString(),
-            'currency_id': this.txContext.currencyId &&
-                           parseInt(this.txContext.currencyId),
-            'partner_id': parseInt(this.txContext.partnerId),
```

Comment:
Why remove that here but add them there? https://github.com/odoo/odoo/pull/116531/files#diff-197f0de1c9f51c575093e9c41cc5f6c8c8afcc75a20c3093cd4a234c18c82b59R87-R91

Reply:
How is the `currency_id` set?

Reply:
Because we don't want them to be passed in general, but passed in case of donation. In most cases currency is set based on document(order_id, invoice_id etc).